### ISSUE TITLE
Assertion on Non-ASCII HeaderDict keys when run with Python 2.x 

### ISSUE THREAD

**Issue by bad32** [bottlepy/bottle#914]
[Python asserts that header names are StringType](https://github.com/python/cpython/blob/b6f258d2a7b22df78ae30ebe2b7910d8d65bb918/Lib/wsgiref/handlers.py#L179), which crashes the code whenever unicode strings are used directly as keys in python 2.x, or when someone uses `unicode_literals` from `__future__` in their python 2/3 compatible code.

Reproducible sample:
```
from __future__ import unicode_literals
import bottle


@bottle.route('/')
def index():
    bottle.response.headers['Content-Encoding'] = ''
    return ''

if __name__ == '__main__':
    bottle.run()

```

possible solution could be normalizing the key returned from [`_hkey`](https://github.com/bottlepy/bottle/blob/f936dfaa7ca43551ba63d9a7b4b2a9a62ab9c0fb/bottle.py#L1586)
```
return str(s.title()).replace('_', '-')
```

**Comment by eric-wieser** [comment]
Does python 3 allow `unicode` headers?

**Comment by bad32** [comment]
Yes, You can run the sample I provided without issue using python3.

**Comment by eric-wieser** [comment]
In fact, python 3 will complain if you use `bytes`, it seems

**Comment by bad32** [comment]
Seems reasonable to support the literal-string type of the running python environment; 
`unicode_literals` changes the type for the caller's module, and not bottle. My proposed change simply enforces the real literal-type that bottle expects.

**Comment by defnull** [comment]
Bottle expects 'native strings' (`str()` type) most of the time. Using `from __future__ import unicode_literals` will probably break a lot more than just headers.
